device-connectors/src/testflinger_device_connectors/devices/muxpi/muxpi.py

```diff
@@ -113,7 +113,53 @@ def _copy_to_control(self, local_file, remote_file):
             raise ProvisioningError(e.output)
         return output
 
+    def check_control_alive(self):
+        """Check if the control host is alive"""
+        try:
+            self._run_control("true")
+        except (ProvisioningError, subprocess.SubprocessError):
+            raise ProvisioningError(
+                "Control host is not responding, provisioning can't proceed!"
+            )
```

Comment:
The question of where to put the exception handling logic depends on the concept of the functions you implement. If you think that `_run_control` should also be responsible for this, you can move this logic to that function. However, in this case I don't see any point of the `check_control_alive`, since it just calls the `_run_control` function, probably we should get rid of it at all
My original suggestion for this piece of code was about a bit different thing. I propose also to replace `raise ProvisioningError(e.output)` with `raise ProvisioningError(e.output) from e`